server/routes_test.go

```diff
@@ -53,6 +53,55 @@ func TestRouteConfig(t *testing.T) {
 	}
 }
 
+func TestClientRemoveSubsOnDisconnectFromCluster(t *testing.T) {
+	optsA, _ := ProcessConfigFile("./configs/srv_a.conf")
+	optsB, _ := ProcessConfigFile("./configs/srv_b.conf")
+
+	optsA.NoSigs, optsA.NoLog = true, true
+	optsB.NoSigs, optsB.NoLog = true, true
+
+	srvA := RunServer(optsA)
+	defer srvA.Shutdown()
+
+	srvB := RunServer(optsB)
+	defer srvB.Shutdown()
+
+	cluster := []*Server{srvA, srvB}
+
+	// Wait for route to form.
+	checkClusterFor(t, routesCheck(len(cluster)-1), 10*time.Second, cluster...)
```

Comment:
I see now that you use this function for different type of checks. However, to check for cluster formation, there is already a function. You could use:`checkClusterFormed(t, srvA, srvB)`. I see that yours is more generic (I did something very similar in NATS Streaming server). Since `checkClusterFormed()` is used quite a bit, we could change `checkClusterFormed()` implementation to simply call into yours. Does not have to be part of this PR though...